### PR TITLE
fix: [ANDROAPP-2823] Fix add event when grouping events

### PR DESCRIPTION
**Pull request by Balcan**
[issue](https://jira.dhis2.org/browse/ANDROAPP-2823)

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/tei_data/tei_events/StageViewHolder.kt [hunk 1]
```diff
@@ -39,8 +39,9 @@ internal class StageViewHolder(
         )
         binding.lastUpdatedEvent.text = eventItem.lastUpdate.toDateSpan(itemView.context)
         binding.addStageButton.visibility =
-            if (stage.repeatable() == true &&
-                eventItem.canAddNewEvent || eventItem.eventCount == 0
+            if (
+                eventItem.canAddNewEvent && (stage.repeatable() != true && eventItem.eventCount == 0 ||
+                        stage.repeatable() == true && eventItem.eventCount == 0 || stage.repeatable() == true && eventItem.eventCount > 0 && eventItem.isSelected)
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/tei_data/tei_events/StageViewHolder.kt]: can you extract this comparisons to different boolean variables for better understanding? thanks :)

Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/tei_data/tei_events/StageViewHolder.kt]: on my way!

Comment (JaimeToca) [reply app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/tei_data/tei_events/StageViewHolder.kt]: thanks

**Comment by ferdyrod** [comment]
Tested!

**Pull request merged**